pkg/tlscert/webhook_observer.go

```diff
@@ -45,31 +48,50 @@ func NewWebhookObserver(kubeClient kubernetes.Interface, namespace string, logge
 		informerMutatingWebhooks:   factory.Admissionregistration().V1().MutatingWebhookConfigurations().Informer(),
 		log:                        logger,
 		stopCh:                     make(chan struct{}),
+		lock:                       sync.Mutex{},
 	}
 
 	return c
 }
 
+func (c *WebhookObserver) onWebHookConfigurationObserved(obj interface{}) {
+
+	if c.onEvent == nil {
+		level.Error(c.log).Log("msg", "webhook configuration listener has not been initialized")
+		return
+	}
```

Comment:
I don't think this is necessary - if `onEvent` is `nil`, something is wrong and panicking is fine.